src/cpp/rtps/RTPSDomain.cpp

```diff
@@ -317,6 +344,108 @@ bool RTPSDomain::removeRTPSReader(
+        return nullptr;
+    }
+
+    // Check for the environment variable
+    #pragma warning(suppress:4996)
+    const char* address = std::getenv(DEFAULT_ROS2_MASTER_URI);
+
+    if( nullptr == address )
+    {
+        // Back to default creation procedure
+        return nullptr;
+    }
+
+    logInfo(DOMAIN, "Detected ROS client-server environment variable. Trying to setup client-server default setup.");
+
+    // Parse the IPv4Address and port.
+    std::cmatch mr;
+    std::string ip_address;
+    int port = DEFAULT_ROS2_SERVER_PORT;
+
+    if(regex_match(address, mr, ROS2_IPV4_PATTERN))
+    {
+        std::cmatch::iterator it = mr.cbegin();
+        ip_address = (++it)->str();
+
+        if((++it)->matched)
+        {
+            port = std::stoi(it->str());
+        }
+    }
+    else
+    {
+        logError(DOMAIN, "The environment variable '" << DEFAULT_ROS2_MASTER_URI
+            << "' has an invalid IPv4 pattern '" << address << "'");
+        return nullptr;
+    }
+
+    // Check if the IP address belong to the local interfaces
+    RTPSParticipant* part = nullptr;
+    LocatorList_t locals;
+    Locator_t server_address(port);
+    IPLocator::setIPv4(server_address, ip_address);
+
+    IPFinder::getIP4Address(&locals);
+    if(locals.end() != std::find_if(locals.begin(), locals.end(),
+        [&server_address](const Locator_t & loc) -> bool
+        {
+            return IPLocator::compareAddress(server_address, loc);
+        }))
+    {
+        RTPSParticipantAttributes server_attr = att;
+        server_attr.builtin.discovery_config.discoveryProtocol = DiscoveryProtocol_t::SERVER;
+        server_attr.ReadguidPrefix(DEFAULT_ROS2_SERVER_GUIDPREFIX);
+        server_attr.builtin.metatrafficUnicastLocatorList.push_back(server_address);
+
+        part = RTPSDomain::createParticipant(domain_id, server_attr, listen);
+        if(nullptr == part)
+        {
+            // There wasn't any previous default server, now there is one
+            logInfo(DOMAIN, "Ros2 default client-server setup. Default server created.");
+            return part;
+        }
+    }
+
+    // Try to create a Server. If it's already a default one the creation process would fail and we will create
+    // a client instead
+    logInfo(DOMAIN, "Ros2 default client-server setup. Server already present trying to create client.");
+
+    // There was a server already or server IP doesn't match this machine. Let's create a client
+    {
+        rtps::RTPSParticipantAttributes client_attr = att;
+        client_attr.builtin.discovery_config.discoveryProtocol = DiscoveryProtocol_t::CLIENT;
+
+        RemoteServerAttributes ratt;
+        ratt.ReadguidPrefix(DEFAULT_ROS2_SERVER_GUIDPREFIX);
+        ratt.metatrafficUnicastLocatorList.push_back(server_address);
+        client_attr.builtin.discovery_config.m_DiscoveryServers.push_back(ratt);
+
+        part = RTPSDomain::createParticipant(domain_id, client_attr, listen);
+        if(nullptr == part)
```

Comment:
```suggestion
        if(nullptr != part)
```

Reply:
You are right! I was asleep for sure when I take the assignment out of the condition to avoid gcc warnings.